exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -18,23 +18,31 @@ package controllers
 
 import (
```

Comment:
General comment(s):
* please go over the errors we currently return and check if we need more error wrapping to provide enough context
* there are a few cases where we should change Extension => ExtensionConfig